internal/kafka/internal/services/quota/ams_quota_service.go

```diff
@@ -295,6 +296,39 @@ func (q amsQuotaService) ReserveQuota(kafka *dbapi.KafkaRequest) (string, *error
 	return resp.Subscription().ID(), nil
 }
 
+func (q amsQuotaService) ReserveQuotaIfNotAlreadyReserved(kafka *dbapi.KafkaRequest) (string, *errors.ServiceError) {
+	instanceType, err := q.kafkaConfig.SupportedInstanceTypes.Configuration.GetKafkaInstanceTypeByID(kafka.InstanceType)
+	if err != nil {
+		return "", errors.GeneralError("failed checking current quota: %v", err)
+	}
+	kafkaBillingModel, err := instanceType.GetKafkaSupportedBillingModelByID(kafka.DesiredKafkaBillingModel)
+	if err != nil {
+		return "", errors.GeneralError("failed checking current quota: %v", err)
+	}
+
+	subscriptions, err := q.amsClient.FindSubscriptions(fmt.Sprintf("cluster_id='%s'", kafka.ClusterID))
+	if err != nil {
+		return "", errors.GeneralError("failed checking current quota: %v", err)
+	}
+	for _, subscription := range subscriptions {
+		// TODO: should we check subscription status?
```

Comment:
I think so. What happens if there's an old subscription that is not used anymore?